plugins/module_utils/network/sonic/config/bgp_communities/bgp_communities.py

```diff
@@ -255,70 +285,92 @@ def get_delete_all_bgp_communities(self, commands):
         return requests
 
     def get_delete_bgp_communities(self, commands, have, is_delete_all):
-        # with open('/root/ansible_log.log', 'a+') as fp:
-        #     fp.write('bgp_commmunities: delete requests ************** \n')
         requests = []
         if is_delete_all:
             requests = self.get_delete_all_bgp_communities(commands)
         else:
             for cmd in commands:
                 name = cmd['name']
-                type = cmd['type']
                 members = cmd['members']
-                if members:
-                    if members['regex']:
-                        diff_members = []
-                        for item in have:
-                            if item['name'] == name and item['members']:
+                aann_list = cmd['aann']
+                local_as = cmd['local_as']
+                no_advertise = cmd['no_advertise']
+                no_export = cmd['no_export']
+                no_peer = cmd['no_peer']
+
+                diff_members = []
+
+                for item in have:
+                    if item['name'] == name:
+                        if local_as and local_as == item['local_as']:
+                            diff_members.append("NO_EXPORT_SUBCONFED")
+                        if no_advertise and no_advertise == item['no_advertise']:
+                            diff_members.append("NO_ADVERTISE")
+                        if no_export and no_export == item['no_export']:
+                            diff_members.append("NO_EXPORT")
+                        if no_peer and no_peer == item['no_peer']:
+                            diff_members.append("NOPEER")
+
+                        if members:
+                            if members['regex']:
                                 for member_want in members['regex']:
                                     if str(member_want) in item['members']['regex']:
-                                        diff_members.append(member_want)
-                        if diff_members:
-                            requests.extend(self.get_delete_single_bgp_community_member_requests(name, type, diff_members))
-                    else:
-                        for item in have:
-                            if item['name'] == name:
-                                if item['members']:
-                                    requests.append(self.get_delete_all_members_bgp_community_requests(name))
-                else:
-                    for item in have:
-                        if item['name'] == name:
-                            requests.append(self.get_delete_single_bgp_community_requests(name))
+                                        diff_members.append("REGEX:" + str(member_want))
+                            else:
+                                for item in have:
+                                    if item['name'] == name:
```

Comment:
These conditions are duplicate of conditions at lines 303-304, and hence can be removed.